### PR TITLE
docs: updating README to the new flame bloc version

### LINKED ISSUE #1695: [flame_bloc] Flame bloc README don't mention the new API.
# What could be improved
The README should mention the new Flame Bloc API

# Why should this be improved
Right the user has no documentation on how to use the new API.

# More information

### PR DESCRIPTION
**Pull request by erickzanardo**
# Description

Updates the README to match the new API.

## Checklist

- [x] The title of my PR starts with a [Conventional Commit] prefix (`fix:`, `feat:`, `docs:` etc).
- [x] I have read the [Contributor Guide] and followed the process outlined for submitting PRs.
- [ ] I have updated/added tests for ALL new/updated/fixed functionality.
- [x] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [ ] I have updated/added relevant examples in `examples`.

## Breaking Change

- [ ] Yes, this is a breaking change.
- [x] No, this is *not* a breaking change.

## Related Issues

Fixes #1695 

[issue database]: https://github.com/flame-engine/flame/issues
[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Flame Style Guide]: https://github.com/flame-engine/flame/blob/main/STYLEGUIDE.md
[Conventional Commit]: https://conventionalcommits.org

### CODE REVIEW

**Comment by erickzanardo** [comment]
cc @felangel

File: packages/flame_bloc/LICENSE [hunk 1]
```diff
@@ -1,6 +1,6 @@
 MIT License
 
-Copyright (c) 2021 Blue Fire
```
Comment (st-pasha) [inline packages/flame_bloc/LICENSE:3]: see https://www.copyrightlaws.com/copyright-symbol-notice-year/#:~:text=Which%20Year%20Should%20You%20Use,to%20the%20public%20without%20restriction.

File: packages/flame_bloc/DEPRECATED_README.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
+# Flame Bloc 🔥🧱
```
Comment (st-pasha) [inline packages/flame_bloc/DEPRECATED_README.md:1]: Perhaps add a comment here about when this deprecated readme can be removed. This way we make sure the file won't linger forever.

**Review by st-pasha — APPROVED** [review]

File: packages/flame_bloc/README.md [hunk 3]
```diff
@@ -1,53 +1,109 @@
...
-game and makes it possible for Flame components to listen to state changes to those blocs/cubits.
+`flame_bloc` offers a simple and natural (as in similar to `flutter_bloc`) way to use blocs and
+cubits inside a `FlameGame`.
+
+## ⚠️  IMPORTANT ⚠️  ##
+This README only covers the new (>= 1.4.0) API of flame bloc, for the previous deprecated API check
+[this](./DEPRECATED_README.md).
+
+For a migration guide from the previous API to the current one,
+[check this article](https://verygood.ventures/blog/flame-bloc-new-api).
```
Comment (felangel) [inline packages/flame_bloc/README.md:11]: 💯

File: packages/flame_bloc/README.md [hunk 4]
```diff
@@ -1,53 +1,109 @@
...
+      ),
+    );
+  }
+}
 ```
 
-To enable the features of `flame_bloc` in your game, you can make your game class inherit from
-`FlameBlocGame`, or if you are already using an enhanced `FlameGame` class (like for example a
-`Forge2DGame`), the `FlameBloc` mixin can be used instead.
+With that, the `Player` component will now have access to our bloc.
```
Comment (felangel) [inline packages/flame_bloc/README.md:37]:
```suggestion
With the above changes, the `Player` component will now have access to our bloc.
```

File: packages/flame_bloc/README.md [hunk 5]
```diff
@@ -1,53 +1,109 @@
...
+          // ...
+        ],
+      ),
+    );
   }
 }
 ```
 
-To have your components listen to state change, the `BlocComponent` mixin can be used.
+Listening to states changes on a component level can be done with two approaches:
```
Comment (felangel) [inline packages/flame_bloc/README.md:65]:
```suggestion
Listening to states changes at the component level can be done with two approaches:
```

File: packages/flame_bloc/README.md [hunk 6]
```diff
@@ -1,53 +1,109 @@
...
   @override
-  void update(double dt) {
-    // the `state` getter can also be used to have
-    // direct access to the current state
-    print(state.weapon);
+  void onNewState(state) {
+    updateGear(state);
+  }
+}
+```
```
Comment (felangel) [inline packages/flame_bloc/README.md:95]: Might want to call out one limitation of the mixin is that a component cannot listen to multiple blocs

**Review by wolfenrain — APPROVED** [review]

File: packages/flame_bloc/README.md [hunk 7]
```diff
@@ -1,53 +1,109 @@
 # Flame Bloc 🔥🧱
 
-`flame_bloc` adds easy access to blocs/cubits that are available on the widget tree to your Flame
-game and makes it possible for Flame components to listen to state changes to those blocs/cubits.
+`flame_bloc` offers a simple and natural (as in similar to `flutter_bloc`) way to use blocs and
+cubits inside a `FlameGame`.
+
+## ⚠️  IMPORTANT ⚠️  ##
+This README only covers the new (>= 1.4.0) API of flame bloc, for the previous deprecated API check
+[this](./DEPRECATED_README.md).
```
Comment (spydon) [inline packages/flame_bloc/README.md:8]: I don't think we need this, if they want to read an old version of the readme they can look at the tag for an old version.

**Review by spydon — COMMENTED** [review]
Looks good, once the requested changes from Felix are applied!
And I don't think that we should keep the deprecated readme file, potentially link to the old tagged version instead if you want the user to find it (but I don't think that is necessary either).

**Pull request merged**